tests/test_atom_mapping.py

```diff
@@ -953,3 +971,281 @@ def test_get_cores_and_diagnostics():
+        "min_threshold": 0,
+        "initial_mapping": initial_mapping,
+    }
+
+    all_cores_test, diagnostics_test = atom_mapping.get_cores_and_diagnostics(mol_a, mol_b, **TEST_ATOM_MAPPING_KWARGS)
+    TEST_ATOM_MAPPING_KWARGS["initial_mapping"] = None
+    all_cores_ref, diagnostics_ref = atom_mapping.get_cores_and_diagnostics(mol_a, mol_b, **TEST_ATOM_MAPPING_KWARGS)
+
+    assert len(all_cores_test[0]) == len(all_cores_ref[0])
+
+    # should be something like "Test visited: 1480 Ref visited: 31796"
+    print("Test visited:", diagnostics_test.total_nodes_visited, "Ref visited:", diagnostics_ref.total_nodes_visited)
+    assert diagnostics_test.total_nodes_visited < diagnostics_ref.total_nodes_visited
+
+
+def new_to_old_map_after_removing_hs(mol):
+    # usually explicitHs are always placed at the end, but just to be safe
+    # we explicitly compute the ordering
+    atomic_nums = [a.GetAtomicNum() for a in mol.GetAtoms()]
+    old_to_new_mapping = {}
+    new_to_old_mapping = {}
+
+    # generate old to new mapping
+    for atom_idx, atom_num in enumerate(atomic_nums):
+        if atom_num != 1:
+            old_to_new_mapping[atom_idx] = len(old_to_new_mapping)
+
+    # generate new to old mapping
+    for old, new in old_to_new_mapping.items():
+        new_to_old_mapping[new] = old
+
+    return new_to_old_mapping
+
+
+def test_empty_initial_mapping_returns_identity(hif2a_ligands):
+    rng = np.random.default_rng(2024)
+    mols = hif2a_ligands
+    mol_a, mol_b = rng.choice(mols, 2)
+
+    kwargs = DEFAULT_ATOM_MAPPING_KWARGS.copy()
+    cores = atom_mapping.get_cores(mol_a, mol_b, **kwargs)
+
+    # Don't change the kwargs here, besides adding an empty core
+    kwargs["initial_mapping"] = np.zeros((0, 2))
+
+    cores_empty_initial_map = atom_mapping.get_cores(mol_a, mol_b, **kwargs)
+
+    assert len(cores) > 1
+    assert len(cores_empty_initial_map) == len(cores)
+
+    np.testing.assert_equal(cores, cores_empty_initial_map)
+
+
+def test_initial_mapping_returns_self_with_same_params(hif2a_ligands):
+    """If an initial map is provided for the same parameters, the values are identical"""
+    rng = np.random.default_rng(2024)
+    mol_a, mol_b = rng.choice(hif2a_ligands, 2)
+
+    kwargs = DEFAULT_ATOM_MAPPING_KWARGS.copy()
+    cores = atom_mapping.get_cores(mol_a, mol_b, **kwargs)
+
+    kwargs["initial_mapping"] = cores[0]
+
+    # Since the initial mapping is set and no parameters are changed, should only return a single core
+    identity_core = atom_mapping.get_cores(mol_a, mol_b, **kwargs)
+
+    assert len(cores) > 1
+    assert len(identity_core) == 1
+    # Core ordering does get shuffled, but pairs should be identical
+    assert set(tuple(pair) for pair in identity_core[0]) == set(tuple(pair) for pair in cores[0])
+
+
+def test_initial_mapping_always_a_subset_of_cores(hif2a_ligands):
+    rng = np.random.default_rng(2024)
+    mols = hif2a_ligands
+    mol_a, mol_b = rng.choice(mols, 2)
```

Comment:
testing observation: these tests are each currently parameterized by a list of ligands `hif2a_ligands`, but run a single random pair from the list.

consider either:
* parameterizing by `(ligands, num_random_pairs=1)`
* parameterizing by `mol_pair`

Reply:
Parameterized the pairs in https://github.com/proteneer/timemachine/pull/1254/commits/78b1b35514448b80f7043dd2a170654b885eb0df